src/components/ListOptions/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { FunctionComponent} from 'react'
+import {IonPopover, IonIcon, IonToolbar, IonButtons, IonButton, IonList, IonListHeader, IonItem, IonLabel, IonRadioGroup, IonRadio} from '@ionic/react'
+import s from './styles.module.scss'
+
+const ListOptions: FunctionComponent = ({ isOpen, onClose }) => {
+  const options = JSON.parse(localStorage.getItem("list-options")) || {
```

Comment:
this is a potentially slow bit of code that is being executed on every re-render (even if nothing has changed in the dom). While I trust you might not "feel" it, this codebase is also educational. This is a great place where we can demo how to optimize calls like this.

Reply:
"list-options" in localStorage might be changed without our knowledge, legitimately - like in the popover, but potentially elsewhere.  We might end up having a "reset" kind of feature that clears localStorage, among other things.